docs/spec/spec.md

```diff
@@ -0,0 +1,1821 @@
+            * 8.1.1.3 [Returning error](#8113-returning-error)
+        * 8.1.2 [Request error interceptor](#812-request-error-interceptor)
+        * 8.1.3 [Engaging interceptor](#813-engaging-interceptors)
+            * 8.1.3.1 [Service level](#8131-service-level)
+            * 8.1.3.2 [Listener level](#8132-listener-level)
+            * 8.1.3.3 [Execution order of interceptors](#8133-execution-order-of-interceptors)
+        * 8.1.4 [Data binding](#814-data-binding)
+        * 8.1.5 [Return to respond](#815-return-to-respond)
+     * 8.2. [Error handling](#82-error-handling)
+        * 8.2.1. [Trace log](#821-trace-log)
+        * 8.2.2. [Access log](#822-access-log)
+9. [Security](#9-security)
+    * 9.1. Authentication and Authorization
+        * 9.1.1. Declarative Approach
+            * 9.1.1.1. Service - File User Store
+            * 9.1.1.2. Service - LDAP User Store
+            * 9.1.1.3. Service - JWT
+            * 9.1.1.4. Service - OAuth2
+            * 9.1.1.5. Client - Basic Auth
+            * 9.1.1.6. Client - Bearer Token Auth
+            * 9.1.1.7. Client - Self Signed JWT Auth
+            * 9.1.1.8. Client - OAuth2
+        * 9.1.2 Imperative Approach
+            * 9.1.2.1. Service - File User Store
+            * 9.1.2.2. Service - LDAP User Store
+            * 9.1.2.3. Service - JWT
+            * 9.1.2.4. Service - OAuth2
+            * 9.1.2.5. Client - Basic Auth
+            * 9.1.2.6. Client - Bearer Token Auth
+            * 9.1.2.7. Client - Self Signed JWT Auth
+            * 9.1.2.8. Client - OAuth2   
+10. [Protocol-upgrade](#10-protocol-upgrade)
+    * 10.1. [HTTP2](#101-http2)
+        * 10.1.1. [Push promise and promise response](#1011-push-promise-and-promise-response)
+    * 10.2. [Websocket](#102-websocket)
+
+## 1. Overview
+This specification elaborates on Basic Auth authentication and authorization for all the Ballerina listeners and
+clients. The HTTP, gRPC, GraphQL, WebSocket, WebSub protocol-based listeners and clients are secured according to this
+specification.
+
+The ballerina language provides the first class support for writing network oriented programming models in a more 
+initiative manner. The ballerina-http standard library consumes the rich language construct and creates the 
+programming model to write services.
+
+## 2. Components
+### 2.1. Listener
+The HTTP listener object receives network data from a remote process according to the HTTP transport protocol and 
+translates the received data into invocations on the resources functions of services that have been 
+attached to the listener object. The listener provides the interface between network and services.
+
+As defined in [Ballerina 2021R1 Section 5.7.4](https://ballerina.io/spec/lang/2021R1/#section_5.7.4) the Listener has 
+the object constructor and life cycle methods such as attach(), detach(), 'start(), gracefulStop(), and immediateStop().
+
+#### 2.1.1. Auto binding
```

Comment:
I feel `binding` is not the correct word here.

Reply:
2.1.1. Automatically starting the service
2.1.2. Programmatically starting the service

Reply:
Above two might be a better option.